x-pack/plugins/fleet/server/services/agents/helpers.ts

```diff
@@ -92,6 +92,7 @@ export function searchHitToAgent(
     // key-value pairs
     user_provided_metadata: hit._source?.user_provided_metadata!,
     local_metadata: hit._source?.local_metadata!,
+    unhealthy_reason: hit._source?.unhealthy_reason,
```

Comment:
added this to make `unhealthy_reason` visible in Agent details UI too:

<img width="1128" alt="image" src="https://github.com/elastic/kibana/assets/90178898/ef266b57-8342-4941-9dda-c70e4c3e42da">